src/core/layertree/qgslayertreemodellegendnode.h

```diff
@@ -303,6 +328,14 @@ class CORE_EXPORT QgsSymbolLegendNode : public QgsLayerTreeModelLegendNode
      */
     void toggleAllItems();
 
+    /**
+     * Evaluates  and returns the text label of the current node
+     * \param context extra QgsExpressionContext to use for evaluating the expression
+     * \param label text to evaluate instead of the layer layertree string
+     * \since QGIS 3.8
+     */
+    QString evaluateLabel( QgsExpressionContext context = QgsExpressionContext(), QString label = QString() );
```

Comment:
Is this a slot on purpose?

Reply:
Yes, otherwise the functionality break from what I recall of the test I did.

Reply:
I can't see why this should be the case. Do you have an error message or explanation?

Reply:
I'll move it in the next commit in the private

Reply:
@m-kuhn see https://travis-ci.org/qgis/QGIS/jobs/525082187, somehow it cannot be access unless its in the public slot. I'll revert it to pass the tests.

Reply:
Just make it public (but not a slot)

Reply:
@m-kuhn This is what I did for the test and it failed.

Reply:
Did you put it into the sip file too?

Reply:
Of course, see https://github.com/qgis/QGIS/commit/a75cd2741997fe1a34b479882209e1d50ac4e2fd
Both were move to the public section in the h and sip file (otherwise sipify would have returned an error).

Reply:
Looks like the `ConvertToSubClassCode` for sip is missing. Add the following snippet to the header. Extra points if you add it for all possible subclasses inside the file.

```
#ifdef SIP_RUN
    SIP_CONVERT_TO_SUBCLASS_CODE
    if ( qobject_cast<QgsLayerTreeModelLegendNode *> ( sipCpp ) )
      sipType = sipType_QgsLayerTreeModelLegendNode;
    else
      sipType = 0;
    SIP_END
#endif
```